Common/src/main/java/com/blamejared/crafttweaker/api/zencode/scriptrun/ScriptRunConfiguration.java

```diff
@@ -0,0 +1,29 @@
+package com.blamejared.crafttweaker.api.zencode.scriptrun;
+
+import com.blamejared.crafttweaker.api.zencode.IScriptLoadSource;
+import com.blamejared.crafttweaker.api.zencode.IScriptLoader;
+import net.minecraft.resources.ResourceLocation;
+
+public record ScriptRunConfiguration(
+        IScriptLoader loader,
+        IScriptLoadSource loadSource,
+        RunKind runKind
+) {
+    
+    public enum RunKind {
+        SYNTAX_CHECK,
+        FORMAT,
+        EXECUTE
+    }
+    
+    public ScriptRunConfiguration(final String loader, final ResourceLocation loadSource, final RunKind runKind) {
+        
+        this(IScriptLoader.find(loader), IScriptLoadSource.find(loadSource), runKind);
+    }
+    
+    public ScriptRunConfiguration(final String loader, final String loadSource, final RunKind runKind) {
```

Comment:
Why this overload?

Reply:
To avoid the `new` on the `ResourceLocation` and make the creation of this object somewhat sleeker. I can remove it if you feel like it does not make a lot of sense